mass_mailing_import_mailing_list/wizard/import_mailing_list.py

```diff
@@ -0,0 +1,360 @@
+                'contact_ids': [(5, )]
+            })
+        list_contact = self.env['mail.mass_mailing.list_contact_rel']
+        for mail_name, contact_id in contacts:
+            mail_list = mailing_lists.filtered(
+                lambda l: l.adkd_mail_name.upper() == mail_name)
+            if mail_list:
+                list_contact_id = list_contact.search(
+                        [('contact_id', '=', contact_id),
+                         ('list_id', '=', mail_list.parent_id.id
+                         if self.import_type == 'adkd'
+                         else mail_list.id)]
+                    )
+                if list_contact_id and list_contact_id.opt_out:
+                    continue
+                mail_list.write({
+                    'contact_ids': [(4, contact_id)],
+                })
+                if self.import_type == 'adkd':
+                    mail_list.parent_id.write({
+                        'contact_ids': [(4, contact_id)],
+                    })
+
+    def import_data(self):
+        """Parse and import a file."""
+        if not self.file or not \
+                self.filename.lower().endswith(
+                    ('.xls', '.xlsx', '.csv', 'txt')):
+            raise UserError(
+                _("Please Select an .xls, .xlsx, "
+                  ".txt or .csv file to Import.")
+            )
+        partner_obj = self.env['res.partner']
+        contact_obj = self.env['mail.mass_mailing.contact']
+        failed_imports = []
+        contacts = []
+        active_mails = set()
+        if self.filename.lower().endswith(('.csv', '.txt')):
+            # Decode data to string, Odoo supplies it as base64
+            csv_data = base64.b64decode(self.file)
+            try:
+                data_file = io.StringIO(csv_data.decode("ISO-8859-1"))
+            except Exception as e:
+                raise UserError('Error %s' % e)
+            # Read CSV
+            headers, *data = csv.reader(data_file, delimiter=';')
+
+            # Verify Header, Force it lowercase and make a dict
+            headers = self.check_header(headers)
+            self.nr_total_rows = len(data)
+            for row in data:
+                try:
+                    active_mail = row[headers['activemail']]
+                    sokande_id = row[headers['sökande id']]
+                    email = row[headers['e-postadress']]
+                except IndexError:
+                    # Empty or malformed row.
+                    _logger.warning(f'Could not parse the row: {row}')
+                else:
+                    # Make list of tuples with (active_mail, contact_id)
+                    contact = self.get_contact(sokande_id,
+                                               partner_obj,
+                                               contact_obj,
+                                               email)
+                    if contact:
+                        contacts.append((active_mail.upper(), contact))
```

Comment:
This might hide that we have duplicates or is it intended to filter away duplicates?